src/commands/lock.rs

```diff
@@ -0,0 +1,122 @@
+//! Implementation of the `lock` subcommand.
+
+use std::collections::HashMap;
+use std::collections::HashSet;
+use std::path::PathBuf;
+
+use anyhow::Result;
+use anyhow::bail;
+use chrono::prelude::*;
+use clap::Parser;
+use crankshaft_docker::Docker as crankshaft_docker;
+use serde::Deserialize;
+use serde::Serialize;
+use wdl::ast::AstToken;
+use wdl::ast::v1::Expr;
+use wdl::ast::v1::LiteralExpr;
+use wdl::cli::Analysis;
+use wdl::cli::analysis::Source;
+
+/// Default name for the lock file.
+const LOCK_FILE: &str = "sprocket.lock";
```

Comment:
I think we should hardcode the filename (`sprocket.lock`), but if the user is supplying a `Source` other than CWD, shouldn't the directory this gets written to be configured via CL arg?

Reply:
I also wouldn't call it the "Default name" as we offer no way to specify something else